src/reports.ts

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  ReportConfig,
+  ReportConfigPage,
+  ReportPage,
+  Response,
+  RulesPageMetadata,
+  Schedule,
+} from "./defs";
+import Errors from "./errors";
+
+/**
+ * @class Reports
+ * Handles interactions with the reports API, including creating, retrieving, updating,
+ * enabling/disabling, deleting, and downloading reports and report configurations.
+ */
+export default class Reports {
+  private readonly contentType: string;
+
+  private readonly reportsEndpoint: string;
+
+  private readonly configsEndpoint: string;
+
+  private readonly reportsUrl: URL;
+
+  /**
+   * @constructor
+   * Initializes the Reports API client.
+   * @param {object} config - Configuration object.
+   * @param {string} config.reportsUrl - Base URL for the reports API.
+   */
+  public constructor({ reportsUrl }: { reportsUrl: string }) {
+    this.reportsUrl = new URL(reportsUrl);
+    this.contentType = "application/json";
+    this.reportsEndpoint = "reports";
+    this.configsEndpoint = "configs";
+  }
+
+  /**
+   * Generates a report using a provided report configuration.
+   * @param {string} domainId - Domain identifier.
```

Comment:
To ensure consistency, let's change these:
```suggestion
   * @param {string} domainId - The unique identifier of the domain.
```